File: Java/Trees/BinarySearchTree.java
public class BinarySearchTree { 

    // Root of Binary Tree 
    Node root; 

    BinarySearchTree() { 
        root = null; 
    } 

    void insert(int key) { 
       root = insertRec(root, key); 
    } 

    // A recursive function to insert a new key 
    // in BST 
    Node insertRec(Node root, int key) { 

        // If the tree is empty, return a new node 
        if (root == null) { 
            root = new Node(key); 
            return root; 
        } 

        // Otherwise, recur down the tree 
        if (key < root.key) 
            root.left = insertRec(root.left, key); 
        else if (key > root.key) 
            root.right = insertRec(root.right, key); 

        // return the (unchanged) node pointer 
        return root; 
    } 

    // This method mainly calls InorderRec() 
    void inorder()  { 
       inorderRec(root); 
    } 

    // A utility function to do inorder traversal of BST 
    void inorderRec(Node root) { 
        if (root != null) { 
            inorderRec(root.left); 
            System.out.print(root.key + " "); 
            inorderRec(root.right); 
        } 
    } 

    // Driver Program to test above functions 
    public static void main(String[] args) { 
       int n,n1;
        Scanner in=new Scanner(System.in);
        BinarySearchTree tree = new BinarySearchTree(); 
        System.out.println("Enter the number of nodes you want:");
        n=in.nextInt();
        System.out.println("Enter the elements:");
        for(int i=1;i<=n;i++)
        {
            n1=in.nextInt();
            tree.insert(n1);//entering the node given by the user
        }
        // print inorder traversal of the BST 
        tree.inorder(); 
    } 
} 

// A binary Tree node 
class Node { 
    int key; 
    Node left, right; 

    public Node(int item) { 
        key = item; 
        left = right = null; 
    } 
}

File: Python/Trees/BinarySearchTree.py
class Node:
    def __init__(self,data):
        self.left = None
        self.right = None
        self.data = data
 

# Insert data into the tree
    def insert(self,data):
        if self.data:
            if data < self.data:
                if self.left is None:
                    self.left = Node(data)
                else:
                    self.left.insert(data)
            elif data > self.data:
                if self.right is None:
                    self.right = Node(data)
                else:
                    self.right.insert(data)
        else:
            self.data = data 

# Print the tree
    def PrintTree(self):
        if self.left:
            self.left.PrintTree()
        print( self.data),
        if self.right:
            self.right.PrintTree()


#Use the insert method to add nodes
n=int(input("Enter the number of nodes you want to enter:"))
num=int(input("Enter a node:"))
root=Node(num)
for i in range(1,n):
    num=input("Enter a node:")
    numint=int(num)
    root.insert(numint)
root.PrintTree()
